aeron-driver/src/main/c/media/aeron_receive_channel_endpoint.c

```diff
@@ -546,6 +558,24 @@ int aeron_receiver_channel_endpoint_validate_sender_mtu_length(
     return 0;
 }
 
+void aeron_receive_channel_endpoint_check_for_re_resolution(
+    aeron_receive_channel_endpoint_t *endpoint,
+    int64_t now_ns,
+    aeron_driver_conductor_proxy_t *conductor_proxy)
+{
+    // MDS is not yet supported in the C media driver,
+    if (endpoint->conductor_fields.udp_channel->has_explicit_control &&
+        now_ns > endpoint->time_of_last_activity_ns + AERON_UDP_DESTINATION_TRACKER_DESTINATION_TIMEOUT_NS)
+    {
+        const char *endpoint_name = endpoint->conductor_fields.udp_channel->uri.params.udp.control;
+        struct sockaddr_storage *addr = &endpoint->conductor_fields.udp_channel->local_control;
```

Comment:
Changed this slightly in the Java version to track the current control address and update on re-resolution change. Same approach can be done here by storing the current sockaddr_storage for the control address in the endpoint.